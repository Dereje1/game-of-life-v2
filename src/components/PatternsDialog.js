import Button from '@mui/material/Button';
import DialogTitle from '@mui/material/DialogTitle';
import DialogContent from '@mui/material/DialogContent';
import DialogActions from '@mui/material/DialogActions';
import Dialog from '@mui/material/Dialog';
import RadioGroup from '@mui/material/RadioGroup';
import Radio from '@mui/material/Radio';
import FormControlLabel from '@mui/material/FormControlLabel';

const options = [
    { value: 'random', label: 'Random' },
    { value: 'blinker', label: 'Blinker' },
    { value: 'toad', label: 'Toad' },
    { value: 'beacon', label: 'Beacon' },
    { value: 'glider', label: 'Glider' },
    { value: 'spaceShip', label: 'Lightweight Space Ship' },
    { value: 'pulsar', label: 'Evolve to Pulsar' },
    { value: 'pentaDecathlon', label: 'Evolve to Penta-Decathlon' },
]

const PatternsDialog = ({
    open,
    handleEntering,
    other,
    radioGroupRef,
    value,
    handleChange,
    handleCancel,
    handleOk

}) => (<Dialog
    sx={{ '& .MuiDialog-paper': { width: '80%', maxHeight: 500 } }}
    maxWidth="xs"
    TransitionProps={{ onEntering: handleEntering }}
    open={open}
    {...other}
>
    <DialogTitle>Choose Pattern</DialogTitle>
    <DialogContent dividers>
        <RadioGroup
            ref={radioGroupRef}
            aria-label="patterns"
            name="patterns"
            value={value}
            onChange={handleChange}
        >
            {options.map(({ value, label }) => (
                <FormControlLabel
                    value={value}
                    key={value}
                    control={<Radio />}
                    label={label}
                />
            ))}
        </RadioGroup>
    </DialogContent>
    <DialogActions>
        <Button autoFocus onClick={handleCancel}>
            Cancel
        </Button>
        <Button onClick={handleOk}>Ok</Button>
    </DialogActions>
</Dialog>)

export default PatternsDialog;