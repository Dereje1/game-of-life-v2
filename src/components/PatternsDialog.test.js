import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PatternsDialog from './PatternsDialog'

let props;
beforeEach(()=>{
    props = {
        open: true,
        value: 'Blinker',
        handleCancel: jest.fn(),
        handleOk: jest.fn(),
        handlePatternChange: jest.fn(),
    }
})

afterEach(()=>{
    props=null
})

test('will render the top control', () => { 
    const wrapper = shallow(<PatternsDialog {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot();
})