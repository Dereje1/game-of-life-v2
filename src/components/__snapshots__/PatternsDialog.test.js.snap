// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render the top control 1`] = `
<ForwardRef(Dialog)
  TransitionProps={
    Object {
      "onEntering": [MockFunction],
    }
  }
  maxWidth="xs"
  open={true}
  sx={
    Object {
      "& .MuiDialog-paper": Object {
        "maxHeight": 530,
        "width": "80%",
      },
    }
  }
>
  <ForwardRef(DialogTitle)>
    Choose Pattern
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)
    dividers={true}
  >
    <ForwardRef(RadioGroup)
      aria-label="patterns"
      name="patterns"
      onChange={[MockFunction]}
      value="Blinker"
    >
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="random"
        label="Random"
        value="random"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="blinker"
        label="Blinker"
        value="blinker"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="toad"
        label="Toad"
        value="toad"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="beacon"
        label="Beacon"
        value="beacon"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="glider"
        label="Glider"
        value="glider"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="spaceShip"
        label="Lightweight Space Ship"
        value="spaceShip"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="pulsar"
        label="Evolve to Pulsar"
        value="pulsar"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="pentaDecathlon"
        label="Evolve to Penta-decathlon"
        value="pentaDecathlon"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="rPentomino"
        label="R-pentomino"
        value="rPentomino"
      />
      <ForwardRef(FormControlLabel)
        control={<ForwardRef(Radio) />}
        key="none"
        label="None"
        value="none"
      />
    </ForwardRef(RadioGroup)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Button)
      autoFocus={true}
      onClick={[MockFunction]}
    >
      Cancel
    </ForwardRef(Button)>
    <ForwardRef(Button)
      onClick={[MockFunction]}
    >
      Ok
    </ForwardRef(Button)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
